front/pages/index.tsx

```diff
@@ -32,12 +32,15 @@ export const getServerSideProps = makeGetServerSidePropsRequirementsWrapper({
       url = `/api/login?inviteToken=${inviteToken}`;
     }
 
-    return {
-      redirect: {
-        destination: url,
-        permanent: false,
-      },
-    };
+    // Don't redirect if we want to explicitly see the product page
+    if (!context.query.show) {
```

Comment:
Let's avoid adding this exception. This function should be straightforward and limit conditional parameters to ensure a safe and predictable outcome as it acts as our primary line of security!

Reply:
? this is not really our primary line of security is it? We could just not redirect at here?

Reply:
Not sure how this impacts security - it's just the redirect to the workspace 
As a user I need to be able to see the content of dust.tt if I really want to without having to logout - so needing 2 outcomes there. 
What do you reckon?

Reply:
Cleaned up to prioritize `inviteToken` presence compared to `show` => inviteToken will now always redirect

Reply:
>? this is not really our primary line of security is it? We could just not redirect at here?

It is in a way that when loading a page, this is the first path tat checks at the user session.

Why do we want this logic to live here, for the website I'd just skip this wrapper (except for the home page that still needs to redirect) and go with the default NextJS wrapper.

Reply:
Removed the ?show logic - not great UX for a logged in user that wants to access the homepage for now, but will do until we move the app to app.dust.tt